pisces/io/response.py

```diff
@@ -697,3 +703,878 @@ def read_pazfir(path, input_samp_rate, calib=None, calper=None, input_units='NM'
+    linesOut: string
+        Line to be added to the file header
+    """
+
+    if inType in ['DISP','VEL','ACC','PRESSURE','TESLA'] and outType in ['VOLTS','COUNTS']:
+        sourceType = 'instrument'
+        if outType == 'COUNTS':
+            linesOut = '# Setting stage {} source description to "instrument" though stage \n# input units are in {} and stage output units are in {}.\n#\n'.format(stageNum, inType, outType)
+        else:
+            linesOut = '# Assuming stage {} source description is "instrument" as stage \n# input units are in {} and stage output units are in {}.\n#\n'.format(stageNum, inType, outType)
+    elif inType == 'VOLTS' and outType == 'VOLTS':
+        sourceType = 'digitizer'
+        linesOut = '# Assuming stage {} source description is "preamplifier" as stage \n# input units and stage output units are in VOLTS.\n#\n'.format(stageNum)
+    elif inType == 'VOLTS' and outType == 'COUNTS':
+        sourceType = 'digitizer'
+        linesOut = '# Assuming stage {} source description is "digitizer" as stage \n# input units are in VOLTS and stage output units are in COUNTS.\n#\n'.format(stageNum)
+    elif inType == 'COUNTS' and outType == 'COUNTS':
+        sourceType = 'anti-alias'
+        linesOut = '# Assuming stage {} source description is "anti-alias" as stage \n# input units and stage output units are in COUNTS.\n#\n'.format(stageNum)
+    else:
+        sourceType = 'unknown'
+        linesOut = '# Setting stage {} source description to "unknown" as stage \n# input and output units of {} and {} are unexpected combination.\n#\n'.format(stageNum, inType, outType)
+
+    return sourceType, linesOut
+
+def get_stationxml_lines(respStage):
+
+    """
+    Extract variables from a response stage to write the contents of that stage to the 
+    file header for documentation purposes.
+
+    Parameters
+    ----------
+    respStage: object
+        Input is ObsPy response object of a finite set of types PolesZerosResponseStage,
+        CoefficientsTypeResponseStage, FIRResponseStage, and ResponseListResponseStage as 
+        well as the ResponseStage parent object.
+
+    Returns
+    -------
+    respStageLines: string
+        Lines to be added to the header describing the input stage object
+    """
+
+    respStageLines = '# Stage {}: Original StationXML Stage\n'.format(respStage.stage_sequence_number)
+    respStageLines = respStageLines + '#\tInput Units: {} ({})\n'.format(respStage.input_units.upper(),respStage.input_units_description)
+    respStageLines = respStageLines + '#\tOutput Units: {} ({})\n'.format(respStage.output_units.upper(),respStage.output_units_description)
+    respStageLines = respStageLines + '#\tStage Gain: {} at {} Hz\n'.format(respStage.stage_gain,respStage.stage_gain_frequency)
+    respStageLines = respStageLines + '#\tStage Type: {} \n'.format(respStage.__class__.__name__)
+    
+    if respStage.__class__.__name__ == 'PolesZerosResponseStage':
```

Comment:
I think this could be written as `if isinstance(respStage, PolesZerosResponseStage): ...` ?

Reply:
yep, that makes more sense, i'll make that adjustment

Reply:
so for some reason, isinstance is not working in this case, so I'm going to leave it as is for the time being...I don't really have time to go in and troubleshoot

Reply:
nevermind figured it out and made the change